src/serializer/ResidualHeapVisitor.js

```diff
@@ -498,8 +499,13 @@ export class ResidualHeapVisitor {
     });
   }
 
-  visitRoots(): void {
-    this.visitGenerator(this.realmGenerator);
+  // Use scopeOverride for setting the global generator to a function's scope
+  // instead of the realm generator.
+  visitRoots(scopeOverride?: Scope): void {
+    if (scopeOverride) this.globalScope = scopeOverride;
```

Comment:
It lets us fallback to the specified scope (in our case a FunctionValue) instead of the realm generator. This should be used to tell the serializer to emit these intrinsic modifications/nested functions/prototypes at the function's scope instead of the global scope.